packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -196,6 +195,8 @@ class _DropdownMenuRouteLayout<T> extends SingleChildLayoutDelegate {
   final _DropdownRoute<T> route;
 
   Rect get buttonRect => route.buttonRect;
+  double get menuTop => route.menuTop;
+  double get menuHeight => route.menuHeight;
   int get selectedIndex => route.selectedIndex;
   ScrollController get scrollController => route.scrollController;
```

Comment:
It seems like all of these values should really come from the State, which should get them from the Route... getting them like this seems very likely to mean we don't always get laid out again when they change. If we somehow guarantee that we do, we should add a comment here saying how that guarantee is made.